walrus-attributes-indexer/src/handlers/blog_post.rs

```diff
@@ -0,0 +1,340 @@
+/// to the committer implementation.
+#[derive(Debug, Clone)]
+pub enum ProcessedWalrusMetadata {
+    Upsert {
+        /// The Blob parent object that owns the Metadata dynamic field. The Object is needed to
+        /// retrieve the Walrus Blob id from its contents.
+        sui_blob_object: Object,
+        /// The ID of the Metadata dynamic field.
+        dynamic_field_id: ObjectID,
+        /// The version of the Metadata dynamic field.
+        df_version: u64,
+        view_count: u64,
+        title: String,
+    },
+    /// Tracks the deletion of a Metadata dynamic field. When committing, this will delete the
+    /// existing row.
+    Delete(ObjectID),
+}
+
+pub struct BlogPostPipeline {
+    metadata_type: StructTag,
+}
+
+impl Processor for BlogPostPipeline {
+    const NAME: &'static str = "blog_post";
+
+    type Value = ProcessedWalrusMetadata;
+
+    /// This pipeline operates on a checkpoint granularity to produce a set of values reflecting the
+    /// state of relevant Metadata dynamic fields at checkpoint boundary.
+    fn process(&self, checkpoint: &Arc<CheckpointData>) -> Result<Vec<Self::Value>> {
+        let checkpoint_input_objects = checkpoint_input_objects(checkpoint)?;
+        let latest_live_output_objects = checkpoint_output_objects(checkpoint)?;
+        // Collect values to be passed to committer. This map is keyed on the dynamic field id.
+        let mut values: BTreeMap<ObjectID, Self::Value> = BTreeMap::new();
+
+        // Process relevant Metadata dynamic fields that were wrapped or deleted in this checkpoint.
+        for (object_id, object) in checkpoint_input_objects.iter() {
+            // If an object appears in both maps, it is still alive at the end of the checkpoint.
+            if latest_live_output_objects.contains_key(object_id) {
+                continue;
+            }
+
+            // Check the checkpoint input state of the Metadata dynamic field to see if it's
+            // relevant to our indexing.
+            let Some((_, _)) = extract_values_and_parent_id(&self.metadata_type, object)? else {
+                continue;
+            };
+
+            // Since the table is keyed on the dynamic field id, this is all the information we need
+            // to delete the correct entry in the commit fn.
+            values.insert(*object_id, ProcessedWalrusMetadata::Delete(*object_id));
+        }
+
+        for (object_id, object) in latest_live_output_objects.iter() {
+            // We only care to emit a record for `Metadata` dynamic fields if they have the "path"
+            // key-value attribute.
+            let Some(((title, view_count), parent_id)) =
+                extract_values_and_parent_id(&self.metadata_type, object)?
+            else {
+                continue;
+            };
+
+            // The parent object must also exist: retrieve it for publisher, blob_id, and other
+            // fields for traceability.
```

Comment:
This is actually not true, because the `Blob` could be wrapped. Would recommend focussing on just indexing the metadata and associating it with a `Blob`, without worrying about trying to also fetch the `Blob` and its owner.